### ISSUE TITLE
SQLFluff does not recognize WITHIN GROUP keyword in LISTAGG function

### ISSUE THREAD

**Issue by wjhrdy** [sqlfluff/sqlfluff#5367]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I was trying to format a SQL query that uses the LISTAGG function with the WITHIN GROUP keyword. SQLFluff does not recognize the WITHIN GROUP keyword and refuses to parse

### Expected Behaviour

SQLFluff should recognize the WITHIN GROUP keyword when used with the LISTAGG function and format the query correctly.

### Observed Behaviour

SQLFluff throws an error when it encounters the WITHIN GROUP keyword in a LISTAGG function. The error message is: "unparsable"

### How to reproduce

1. Create a .sql file with the following query:
    ```sql
    SELECT LISTAGG(name, ',') 
    WITHIN GROUP (ORDER BY name) 
    FROM (VALUES ('foo'), ('bar'), ('baz'), ('foo')) AS t (name);
    ```
3. Run SQLFluff with the command: sqlfluff fix yourfile.sql
4. Observe the error message.

### Dialect

I used the Trino dialect.

### Version

sqlfluff, version 2.3.2

### Configuration
```
[sqlfluff]
templater = dbt
dialect = trino
rules = core
max_line_length = 120
# Comma separated list of rules to exclude, or None
# See https://docs.sqlfluff.com/en/stable/configuration.html#enabling-and-disabling-rules
# AM04 (ambiguous.column_count) and ST06 (structure.column_order) are
# two of the more controversial rules included to illustrate usage.
exclude_rules = ambiguous.column_count, structure.column_order

[sqlfluff:templater:dbt]
project_dir = ./
profiles_dir = ./
profile = local
target = dev
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:indentation]
indented_joins = True
tab_space_size = 4
indented_using_on = True
indent_unit = space
template_blocks_indent = True
allow_implicit_indents = True

# The default configuration for aliasing rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
[sqlfluff:rules:aliasing.table]
aliasing = explicit
[sqlfluff:rules:aliasing.column]
aliasing = explicit

# The default configuration for capitalisation rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
# Typically we find users rely on syntax highlighting rather than
# capitalisation to distinguish between keywords and identifiers.
# Clearly, if your organisation has already settled on uppercase
# formatting for any of these syntax elements then set them to "upper".
# See https://stackoverflow.com/questions/608196/why-should-i-capitalize-my-sql-keywords-is-there-a-good-reason
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
